internal/cli/runner_install.go

```diff
@@ -0,0 +1,259 @@
+package cli
+
+import (
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/installutil"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	"github.com/hashicorp/waypoint/internal/runnerinstall"
+	"github.com/hashicorp/waypoint/pkg/server"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/hashicorp/waypoint/pkg/serverconfig"
+	"github.com/posener/complete"
+	"sort"
+	"strings"
+)
+
+type RunnerInstallCommand struct {
+	*baseCommand
+
+	platform            string `hcl:"platform,required"`
+	adopt               bool   `hcl:"adopt,optional"`
+	serverUrl           string `hcl:"server_url,required"`
+	serverCookie        string `hcl:"server_cookie,optional"`
+	id                  string `hcl:"id,optional"`
+	serverTls           bool   `hcl:"server_tls,optional"`
+	serverTlsSkipVerify bool   `hcl:"server_tls_skip_verify,optional"`
+	serverRequireAuth   bool   `hcl:"server_require_auth,optional"`
+}
+
+func (c *RunnerInstallCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *RunnerInstallCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *RunnerInstallCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:   "platform",
+			Usage:  "Platform to install the Waypoint runner into.",
+			Target: &c.platform,
```

Comment:
It'd be neat if this flag was an `EnumVar`. Then we could fail fast if they provided a unsupported platform in the flag parse step.